### PR TITLE
Upgrade to surefire/failsafe plugins 3.0.0-M8

### PR DESCRIPTION
**Pull request by JiriOndrusek**
Fixes: https://github.com/apache/camel-quarkus/issues/723

[ ] An issue should be filed for the change unless this is a trivial change (fixing a typo or similar). One issue should ideally be fixed by not more than one commit and the other way round, each commit should fix just one issue, without pulling in other changes.
[ ] Each commit in the pull request should have a meaningful and properly spelled subject line and body. Copying the title of the associated issue is typically enough. Please include the issue number in the commit message prefixed by #.
[ ] The pull request description should explain what the pull request does, how, and why. If the info is available in the associated issue or some other external document, a link is enough.
[ ] Phrases like Fix #<issueNumber> or Fixes #<issueNumber> will auto-close the named issue upon merging the pull request. Using them is typically a good idea.
[ ] Please run mvn process-resources -Pformat (and amend the changes if necessary) before sending the pull request.
[ ] Contributor guide is your good friend: https://camel.apache.org/camel-quarkus/latest/contributor-guide.html

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -119,7 +119,7 @@
         <maven-jar-plugin.version>3.2.0</maven-jar-plugin.version>
         <maven-release-plugin.version>2.5.3</maven-release-plugin.version>
         <!-- NOTE: We pin to this version due to https://github.com/apache/camel-quarkus/issues/723 -->
```
Comment (jamesnetherton) [inline pom.xml:121]: I think we can safely remove the NOTE comment now

**Comment by ppalaga** [comment]
@jamesnetherton do you have any idea how to ensure that all tests are executed (beyond checking known suspects manually)?

**Comment by JiriOndrusek** [comment]
@ppalaga It could be possible to manually check reports from previous builds and compare it with the current one.

**Comment by ppalaga** [comment]
Maybe we could:

* Downloading full CI logs before and after
* Grep the logs for test summaries
* Comparing the greps should show zero diffs

WDYT?

**Comment by jamesnetherton** [comment]
> Grep the logs for test summaries

Yeah, I think that would work.

File: pom.xml (hunk 1 above)
Comment (JiriOndrusek) [reply pom.xml:121]: I'll do it

**Comment by JiriOndrusek** [comment]
There were some failing test on CI  after upgrade to surefire-plugin 3.0.0-M5. Problem was caused by using JPMS during execution of tests. **There is a different behavior in default test executions on Oracle JDK and AdoptOpenJDK.** Java modular system is by default turned on (since https://issues.apache.org/jira/browse/SUREFIRE-1570), but there is an option to turn it off (https://issues.apache.org/jira/browse/SUREFIRE-1531). This PR turns off modular system for both JVM and native tests

**Comment by JiriOndrusek** [comment]
I've reported an issue on surefire about different behavior in JPMS with different Java vendors - https://issues.apache.org/jira/browse/SUREFIRE-1809

File: pom.xml [hunk 2]
```diff
@@ -558,6 +561,7 @@
                         <groupId>org.apache.maven.plugins</groupId>
                         <artifactId>maven-surefire-plugin</artifactId>
                         <configuration>
+                            <useModulePath>false</useModulePath>
```
Comment (ppalaga) [inline pom.xml:564]: I do not think this line is required given that there is `<useModulePath>false</useModulePath>` in the pluginManagement section.

**Comment by ppalaga** [comment]
> I've reported an issue on surefire about different behavior in JPMS with different Java vendors - https://issues.apache.org/jira/browse/SUREFIRE-1809

Good work, thanks!

Is my understanding right that any resolution of SUREFIRE-1809 will not impact us, because useModulePath's default is true and we want false anyway? SUREFIRE-1809 is just about a broken autodetection that we do not want to rely on, right?

**Comment by JiriOndrusek** [comment]
@ppalaga yes. I understand it in the sane way. There will be no impact by fix of SUREFIRE-1809, because we are overriding default behavior to different one. It is reported only for the sake of surefire plugin.

**Comment by ppalaga** [comment]
I have approved, but actually have you checked whether the same set of tests is executed before and after this change?

> * Downloading full CI logs before and after
> * Grep the logs for test summaries
> * Comparing the greps should show zero diffs

**Comment by JiriOndrusek** [comment]
I'll verify it. It makes sense because the original issue was about ignored tests.

**Comment by JiriOndrusek** [comment]
Please do not merge this PR. 
There are some differences between execution with surefire 3.0.0-M5 and 2.22.2
I'm switching back to draft PR and I'll try to fix it.

**Comment by JiriOndrusek** [comment]
There is another difference in test summaries between 2.22.2 and 3.0.0-M5 -> reported as https://issues.apache.org/jira/browse/SUREFIRE-1812

**Comment by Tibor17** [comment]
Can you show me your logs?
Did the plugin crash or how did it behave with this error?

**Comment by JiriOndrusek** [comment]
Restested and commented here: https://issues.apache.org/jira/browse/SUREFIRE-1812?focusedCommentId=17276293&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17276293

**Comment by JiriOndrusek** [comment]
Ab issue in maven-surefire plugin is confirmed, fix won't make it into M6 - https://issues.apache.org/jira/browse/SUREFIRE-1812?focusedCommentId=17279194&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17279194

**Comment by JiriOndrusek** [comment]
Problem reported  as https://issues.apache.org/jira/browse/SUREFIRE-1812 for version 3.0.0-M5 does not happen on 3.0.0-M6. I'm going to compare results on cq for current surefire plugin and for 3.0.0-M8.

**Comment by JiriOndrusek** [comment]
There is another problem with 3.0.0-M8, where skipped counts are not correct when @ParametrizedTest is used  (worked correctly in 2.22.2 or 3.0.0-*M7). Reported as https://issues.apache.org/jira/browse/SUREFIRE-2143

**Comment by Tibor17** [comment]
Hi Jiri,

I am aware of such issue SUREFIRE-2143
<https://issues.apache.org/jira/browse/SUREFIRE-2143>. It was never
working. We are reworking the code in many places which would fix several
issues at once. It will be a big change.

**Comment by aldettinger** [comment]
It should be fixed with surefire 3.0.0-M10.

**Comment by JiriOndrusek** [comment]
The newest version is [surefire-3.1.2](https://github.com/apache/maven-surefire/releases/tag/surefire-3.1.2). It's worth a try with that version.

**Comment by jamesnetherton** [comment]
Superseded by #5053.

**Pull request closed without merging**